backend/src/apiserver/storage/pipeline_store_kubernetes_test.go

```diff
@@ -0,0 +1,496 @@
+	require.Equalf(t, pageSize, 1, "List size should not be zero")
+
+	options, err1 = list.NewOptionsFromToken(npt, 1)
+	pipelines, _, _, err3 := store.ListPipelines(&model.FilterContext{}, options)
+	require.Nil(t, err3, "Failed to list pipelines: %v")
+	require.Equalf(t, pipelines[0].Name, "Test Pipeline 1", "Pagination failed")
+}
+
+func TestListK8sPipelines_Pagination_LessThanPageSize(t *testing.T) {
+	store := NewPipelineStoreKubernetes(getClient())
+
+	options, err1 := list.NewOptions(&model.Pipeline{}, 10, "", nil)
+	require.Nil(t, err1, "Failed to create list options: %v")
+
+	pipelines, pageSize, _, err := store.ListPipelines(&model.FilterContext{}, options)
+	require.Nil(t, err, "Failed to list pipelines: %v")
+	require.Equalf(t, pageSize, 1, "Page size should be 1")
+	require.Equalf(t, len(pipelines), 1, "List size should be 1")
+}
+
+func TestGetK8sPipeline(t *testing.T) {
+	// This is important for getting a K8s pipeline
+	viper.Set("POD_NAMESPACE", "Test")
+	store := NewPipelineStoreKubernetes(getClient())
+
+	pipeline := &model.Pipeline{
+		UUID:        DefaultFakePipelineId,
+		Name:        "Test Pipeline 3",
+		Description: "Test Pipeline 3 Description",
+		Namespace:   "Test",
+	}
+
+	p, err := store.GetPipeline(pipeline.UUID)
+	require.Nil(t, err, "Failed to get Pipeline: %v", err)
+	require.Equal(t, p.UUID, pipeline.UUID)
+}
+
+func TestGetK8sPipeline_NotFoundError(t *testing.T) {
+	// This is important for getting a K8s pipeline
+	viper.Set("POD_NAMESPACE", "Test")
+	store := NewPipelineStoreKubernetes(getClient())
+
+	pipeline := &model.Pipeline{
+		UUID:        DefaultFakePipelineIdTwo,
+		Name:        "Test Pipeline 3",
+		Description: "Test Pipeline 3 Description",
+		Namespace:   "Test",
+	}
+
+	_, err := store.GetPipeline(pipeline.UUID)
+	require.NotNil(t, err)
+}
+
+func TestCreateK8sPipeline(t *testing.T) {
+	store := NewPipelineStoreKubernetes(getClient())
+
+	pipeline := &model.Pipeline{
+		Name:        "Test Pipeline",
+		Description: "Test Pipeline Description",
+		Namespace:   "Test",
+	}
+
+	pipeline, err := store.CreatePipeline(pipeline)
+	if err != nil {
+		t.Fatalf("Failed to create Pipeline: %v", err)
+	}
+
+	require.Equalf(t, pipeline.Name, "Test Pipeline", "Pipeline name is not the same")
+}
+
+func TestDeleteK8sPipeline(t *testing.T) {
+	store := NewPipelineStoreKubernetes(getClient())
+
+	err := store.DeletePipeline(DefaultFakePipelineId)
+	require.Nil(t, err, "Failed to delete Pipeline: %v", err)
+
+	// Check if Deletion worked by querying the same UUID
+	_, err1 := store.GetPipeline(DefaultFakePipelineId)
+	require.NotNil(t, err1)
+}
+
+func TestCreateK8sPipelineVersion(t *testing.T) {
+	store := NewPipelineStoreKubernetes(getClient())
+
+	pipelineVersion := &model.PipelineVersion{
+		Name:        "Test Pipeline Version",
+		PipelineId:  DefaultFakePipelineId,
+		Description: "Test Pipeline Version Description",
+	}
+
+	pipelineVersion, err := store.CreatePipelineVersion(pipelineVersion)
```

Comment:
Could you please add an assertion that the object was created in the format you expected in Kubernetes (from `getClient()`? Any other tests that test create, updates, and deletes should also do this. This is because we don't have direct test coverage of `FromPipelineVersionModel` and `ToModel`.